src/ngraph/ngraph_emitter.cc

```diff
@@ -1038,11 +1038,22 @@ void Emitter::CreateLayerOps() {
     const NgraphNodePtr one = makeConstant(dtype, shape, 1.0f);
 
     NgraphNodePtr scale_grad;
+
+    #if MXNET_USE_NGRAPH_DISTRIBUTED
+    NgraphNodePtr ng_grad = std::make_shared<ngraph::op::AllReduce>(grad);
```

Comment:
Let's de-constify grad so you can simply overwrite it here and avoid the code copy.